lib/internal/worker.js

```diff
@@ -261,10 +264,24 @@ class Worker extends EventEmitter {
     // Actually start the new thread now that everything is in place.
     this[kHandle].startThread();
 
+    if (~~options.timeout > 0) {
+      this.#timer = setTimeout(() => {
+        this.#timer = null;
+        this.emit('timeout');
+        this.terminate();
+      }, ~~options.timeout).unref();
+    }
+
     process.nextTick(() => process.emit('worker', this));
   }
-
+  clearTimer() {
```

Comment:
Should this method be private?

Reply:
I add this method to docs then user can cancel the timer.